### PR TITLE
feat(condo): INFRA-466 improve syncTourSteps service

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
- check that step is not completed before query step related items
- query first one object instead count

### CODE REVIEW

File: apps/condo/domains/onboarding/schema/SyncTourStepsService.js [hunk 1]
```diff
@@ -41,52 +48,52 @@ const SyncTourStepsService = new GQLCustomSchema('SyncTourStepsService', {
...
+                const notCompletedSteps = tourSteps.filter(step => step.status !== COMPLETED_STEP_STATUS)
 
-                for (const step of tourSteps) {
-                    if (step.status !== 'completed' && stepTypeToCount[step.type]) {
-                        await TourStep.update(context, step.id, {
-                            status: 'completed',
+                for (const { type, id } of notCompletedSteps) {
+                    if (!stepToItemsQuery[type]) continue
+
+                    const objs = await itemsQuery(stepToItemsQuery[type].list, {
```
[the remark this answers is not in the document]
Comment (nomerdvadcatpyat) [reply apps/condo/domains/onboarding/schema/SyncTourStepsService.js:89]: itemsQuery it's our utility

File: apps/condo/k6/src/ticket.test.ts [hunk 2]
```diff
@@ -55,10 +55,10 @@ export const options = {
     },
     thresholds: {
         http_req_failed: ['rate<0.01'],
-        browser_http_req_duration: ['p(95) < 4000', 'med < 1500'],
+        // browser_http_req_duration: ['p(95) < 4000', 'med < 1500'],
         http_req_duration: ['p(95)<3000', 'med<2000'],
-        browser_web_vital_fcp: ['p(95) < 8000', 'med < 4000'],
-        browser_web_vital_lcp: ['p(95) < 15000', 'med < 11000'],
+        // browser_web_vital_fcp: ['p(95) < 8000', 'med < 4000'],
```
Comment (toplenboren) [inline apps/condo/k6/src/ticket.test.ts:60]: //

Comment (nomerdvadcatpyat) [reply apps/condo/k6/src/ticket.test.ts]: fixed

File: apps/condo/k6/src/syncTourStep.test.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    const data = {
+        token,
+        organizationId,
+        propertyId: createdProperty.json('data.obj.id'),
+    }
+
+    for (let i = 0; i < 200; i++) {
+        createTicket(data)
+        createMeter(data)
+    }
```
Comment (Alllex202) [inline apps/condo/k6/src/syncTourStep.test.ts:49]: Should you add a billing entities test here? This seems was the request that took the longest to process.

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/domains/onboarding/schema/SyncTourStepsService.js [hunk 4]
```diff
@@ -2,11 +2,18 @@
...
 
+const isEmpty = require('lodash/isEmpty')
+
 const { GQLError } = require('@open-condo/keystone/errors')
 const { GQLCustomSchema, find, itemsQuery } = require('@open-condo/keystone/schema')
 
 const access = require('@condo/domains/onboarding/access/SyncTourStepsService')
 const { SYNC_TOUR_STEPS_ERRORS } = require('@condo/domains/onboarding/constants/errors')
+const {
+    CREATE_PROPERTY_STEP_TYPE, CREATE_PROPERTY_MAP_STEP_TYPE, CREATE_TICKET_STEP_TYPE,
```
Comment (SavelevMatthew) [inline apps/condo/domains/onboarding/schema/SyncTourStepsService.js:13]: (optional) It's usual 1 per line, or grouped by logic per line, or all in 1 line. A bit strange. But just a tip for future, not requirement

File: apps/condo/k6/src/syncTourStep.test.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            executor: 'constant-arrival-rate',
+            duration: DURATION,
+            rate: 5,
+            timeUnit: '1s',
+            preAllocatedVUs: 7,
+        },
+    },
+    thresholds: {
+        http_req_failed: ['rate<0.01'],
+        http_req_duration: ['med<2000'],
```
Comment (SavelevMatthew) [inline apps/condo/k6/src/syncTourStep.test.ts:27]: 2 sec threshold? Isn't it too much?

will It be decreased later?

**Review by SavelevMatthew — APPROVED** [review]
lgtm

File: apps/condo/domains/onboarding/schema/SyncTourStepsService.js [hunk 6]
```diff
@@ -41,52 +53,52 @@ const SyncTourStepsService = new GQLCustomSchema('SyncTourStepsService', {
...
-                    createMeterReadings: hasMeterReadings,
-                    createNews: hasNews,
                 }
+                const notCompletedSteps = tourSteps.filter(step => step.status !== COMPLETED_STEP_STATUS)
 
-                for (const step of tourSteps) {
-                    if (step.status !== 'completed' && stepTypeToCount[step.type]) {
-                        await TourStep.update(context, step.id, {
-                            status: 'completed',
+                for (const { type, id } of notCompletedSteps) {
```
Comment (pahaz) [inline apps/condo/domains/onboarding/schema/SyncTourStepsService.js:91]: may be we can use butch here?

**Pull request merged**